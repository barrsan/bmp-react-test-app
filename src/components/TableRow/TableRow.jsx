import React from 'react';
import PropTypes from 'prop-types';


const propTypes = {
  name: PropTypes.string,
  place: PropTypes.string,
  organization: PropTypes.string,
  description: PropTypes.string,
  scheme: PropTypes.array.isRequired,
};

const defaultProps = {
  name: '',
  place: '',
  organization: '',
  description: '',
};


const TableRow = ({
  name,
  place,
  organization,
  description,
  scheme,
}) => {
  const data = {
    name,
    place,
    organization,
    description,
  };

  return (
    <tr>
      {scheme.map(i => <td key={i.accessor}>{data[i.accessor]}</td>)}
    </tr>
  );
};


TableRow.propTypes = { ...propTypes };
TableRow.defaultProps = { ...defaultProps };


export default TableRow;
