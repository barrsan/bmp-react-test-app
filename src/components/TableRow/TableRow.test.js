import React from 'react';
import { shallow } from 'enzyme';
import TableRow from './TableRow';


describe('TableRow component', () => {
  const props = {
    name: 'Test name',
    place: 'Test place',
    organization: '',
    scheme: [
      {
        header: 'Test 1',
        accessor: 'name',
      },
      {
        header: 'Test 2',
        accessor: 'organization',
      },
      {
        header: 'Test 3',
        accessor: 'place',
      },
    ],
  };

  const wrapper = shallow(<TableRow {...props} />);

  test('renders correctly', () => expect(wrapper).toMatchSnapshot());

  test('renders without crashing', () => {
    expect(wrapper).toHaveLength(1);
  });

  test('renders columns according to the scheme', () => {
    expect(wrapper.find('td').at(0).text()).toBe(props.name);
    expect(wrapper.find('td').at(1).text()).toBe(props.organization);
    expect(wrapper.find('td').at(2).text()).toBe(props.place);
  });
});
