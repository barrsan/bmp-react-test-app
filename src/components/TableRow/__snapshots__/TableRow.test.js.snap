// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TableRow
    description=""
    name="Test name"
    organization=""
    place="Test place"
    scheme={
        Array [
            Object {
              "accessor": "name",
              "header": "Test 1",
            },
            Object {
              "accessor": "organization",
              "header": "Test 2",
            },
            Object {
              "accessor": "place",
              "header": "Test 3",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <td>
          Test name
</td>,
        <td />,
        <td>
          Test place
</td>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "name",
        "nodeType": "host",
        "props": Object {
          "children": "Test name",
        },
        "ref": null,
        "rendered": "Test name",
        "type": "td",
      },
      Object {
        "instance": null,
        "key": "organization",
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "td",
      },
      Object {
        "instance": null,
        "key": "place",
        "nodeType": "host",
        "props": Object {
          "children": "Test place",
        },
        "ref": null,
        "rendered": "Test place",
        "type": "td",
      },
    ],
    "type": "tr",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <td>
            Test name
</td>,
          <td />,
          <td>
            Test place
</td>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "name",
          "nodeType": "host",
          "props": Object {
            "children": "Test name",
          },
          "ref": null,
          "rendered": "Test name",
          "type": "td",
        },
        Object {
          "instance": null,
          "key": "organization",
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "td",
        },
        Object {
          "instance": null,
          "key": "place",
          "nodeType": "host",
          "props": Object {
            "children": "Test place",
          },
          "ref": null,
          "rendered": "Test place",
          "type": "td",
        },
      ],
      "type": "tr",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
